SeaPublicWebsite.Data/DataStores/PropertyDataStore.cs

```diff
@@ -0,0 +1,54 @@
+﻿using SeaPublicWebsite.Data.ErrorHandling;
+using SeaPublicWebsite.Data.Helpers;
+
+namespace SeaPublicWebsite.Data.DataStores;
+
+public class PropertyDataStore
+{
+    private readonly IDataAccessProvider dataAccessProvider;
+
+    public PropertyDataStore(IDataAccessProvider dataAccessProvider)
+    {
+        this.dataAccessProvider = dataAccessProvider;
+    }
+    public async Task<PropertyData> LoadPropertyData(string reference)
+    {
+        if (!await IsReferenceValid(reference))
+        {
+            throw new PropertyReferenceNotFoundException
+            {
+                Reference = reference
+            };
+        }
+        return await dataAccessProvider.GetSinglePropertyData(reference.ToUpper());
+    }
+
+    public async Task<bool> IsReferenceValid(string reference)
+    {
+        var allPropertyData = await dataAccessProvider.GetAllPropertyData();
+        return allPropertyData.Any(p => p.Reference == reference.ToUpper());
```

Comment:
We probably need an index on the Reference column so that this doesn't get horribly slow

Reply:
Also, just noticed that `GetAllPropertyData()` materialises the entire propertyData table into memory which is a bad thing (TM)